addons/stock/models/stock_move_line.py

```diff
@@ -535,14 +535,37 @@ def _get_similar_move_lines(self):
 
     def _create_and_assign_production_lot(self):
         """ Creates and assign new production lots for move lines."""
-        lot_vals = [{
-            'company_id': ml.move_id.company_id.id,
-            'name': ml.lot_name,
-            'product_id': ml.product_id.id,
-        } for ml in self]
+        lot_vals = []
+        grouped_mls_tracked_by_lot = defaultdict(lambda: self.env['stock.move.line'])
+        mls_tracked_by_sn = self.env['stock.move.line']
+        for ml in self:
+            if ml.product_id.tracking == 'lot':
```

Comment:
Why only accept lot and not serial ?

Reply:
Because the issue happen only for lots as you can't have the same serials on different move lines.

Reply:
Yep right. Note that my proposal is protected againt duplicated SN (because the `check_quantity` is done after)